File: donation-system-web/src/types/itemTemplate.ts
export interface ItemTemplateResponse {
  id: string;
  categoryId: string;
  categoryName: string;
  needGroupId: string;
  needGroupName: string;
  name: string;
  isPerishable: boolean;
  requiresRefrigeration: boolean;
  suitableForAutoSuggestion: boolean;
  requiresManualAnalysis: boolean;
  defaultUnitOfMeasure?: string;
  referenceQuantity?: number;
  notes?: string;
  active: boolean;
  createdAt: string;
  updatedAt?: string;
}

export interface CreateItemTemplateRequest {
  categoryId: string;
  needGroupId: string;
  name: string;
  isPerishable: boolean;
  requiresRefrigeration: boolean;
  suitableForAutoSuggestion: boolean;
  requiresManualAnalysis: boolean;
  defaultUnitOfMeasure?: string;
  notes?: string;
  active: boolean;
}

export interface UpdateItemTemplateRequest {
  categoryId: string;
  needGroupId: string;
  name: string;
  isPerishable: boolean;
  requiresRefrigeration: boolean;
  suitableForAutoSuggestion: boolean;
  requiresManualAnalysis: boolean;
  defaultUnitOfMeasure?: string;
  notes?: string;
  active: boolean;
}

File: APISolidarityManager/APISolidarityManager/Entities/ItemTemplate.cs
namespace APISolidarityManager.Models
{
    public class ItemTemplate
    {
        public Guid Id { get; set; }
        public Guid CategoryId { get; set; }
        public Guid NeedGroupId { get; set; }
        public string Name { get; set; } = null!;
        public bool IsPerishable { get; set; }
        public bool RequiresRefrigeration { get; set; }
        public bool SuitableForAutoSuggestion { get; set; }
        public bool RequiresManualAnalysis { get; set; }
        public string? DefaultUnitOfMeasure { get; set; }
        public decimal? ReferenceQuantity { get; set; }
        public string? Notes { get; set; }
        public bool Active { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public ItemCategory Category { get; set; } = null!;
        public NeedGroup NeedGroup { get; set; } = null!;

        public ICollection<Item> Items { get; set; } = new List<Item>();
    }
}